### PR TITLE
Adapt e2e tests to the new foundry-rpc

### PR DESCRIPTION
**Pull request by MSNTCS**

### CODE REVIEW

File: test/package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "codechain-sdk": "^2.0.0-alpha.2",
     "codechain-stakeholder-sdk": "https://github.com/majecty/codechain-stakeholder-sdk-js#v2.0.0-alpha.1",
     "elliptic": "^6.4.1",
+    "foundry-rpc": "github:MSNTCS/foundry-rpc-js#v2.0.0-alpha.1",
```
Comment (majecty) [inline test/package.json]: What do you think about creating a branch named "alpha" in the CodeChain-io/foundry-rpc repository.

File: test/src/e2e/account.test.ts [hunk 2]
```diff
@@ -95,107 +94,100 @@ describe("account", function() {
                     message,
                     secret
                 );
+
```
Comment (majecty) [inline test/src/e2e/account.test.ts]: You commit contains unrelated changes. Please be careful when you create a commit.

File: test/package.json (hunk 1 above)
Comment (MSNTCS) [reply test/package.json]: I followed the same name pattern that you used for `codechain-stakeholder-sdk-js`.

Comment (majecty) [reply test/package.json]: The `codechain-stakeholder-sdk-js`'s version came from the CodeChain version. We are creating a new rpc project for Foundry. Since Foundry is not a stable yet, it does not have a version. So I thought that we don't need the version for foundry-rpc yet.

Comment (majecty) [reply test/package.json]: Releasing a program with a version of greater than 1.0.0 ensures the users that the program is stable. Our foundry, however, is not stable yet.

Comment (MSNTCS) [reply test/package.json]: I will change it.

File: test/src/e2e/account.test.ts [hunk 3]
```diff
@@ -210,4 +200,4 @@ describe("account", function() {
             await node.clean();
         });
     });
-});
+});
```
Comment (majecty) [inline test/src/e2e/account.test.ts]: It seems that your IDE or editor is not following the .editorconfig file.
Since in our `.editorconfig` file has the `insert_final_new_line = true` line in its file, this final new line should not be deleted.
https://github.com/CodeChain-io/foundry/blob/master/test/.editorconfig#L3

Please check this [link](https://github.com/kodebox-io/kodebox-internal/wiki/EditorConfig)

File: test/src/e2e/chain.test.ts [hunk 4]
```diff
@@ -252,42 +242,21 @@ describe("chain", function() {
...
-                expect(result).not.to.be.null;
-                H256.ensure(result);
-            });
-
-        await node.sdk.rpc
-            .sendRpcRequest("chain_getShardRoot", [10000, null])
-            .then(result => {
-                expect(result).to.be.null;
-            });
+        expect(await node.rpc.chain.getShardOwners({shardId: 0, blockNumber: null})).not.to.be.null;
```
Comment (majecty) [inline test/src/e2e/chain.test.ts]: What this line is doing is different from the previous behavior. You should call `H256.ensure` for the return value of `getShardOwners`.

File: test/src/e2e/customAction.test.ts [hunk 5]
```diff
@@ -31,7 +31,9 @@ chai.use(chaiAsPromised);
 const expect = chai.expect;
 
 const RLP = require("rlp");
-
+const hitcount = toHex(RLP.encode("hit count"));
+const closecount = toHex(RLP.encode("close count"));
+const nonexistingkey  = toHex(RLP.encode("non-existing-key"));
```
Comment (majecty) [inline test/src/e2e/customAction.test.ts]: Please keep the line ending.

File: test/src/e2e/customAction.test.ts [hunk 6]
```diff
@@ -42,27 +44,32 @@ describe("customAction", function() {
...
-                hitActionHandlerId,
-                ["hit count"]
+            const actionData = await node.rpc.engine.getCustomActionData({
+                    handlerId: hitActionHandlerId,
+                    bytes: `0xca${hitcount}`,
+                    blockNumber: null
+                }
             );
-
             expect(actionData).to.be.equal(toHex(RLP.encode(1)));
```
Comment (majecty) [inline test/src/e2e/customAction.test.ts]: Please keep the line ending.

**Comment by MSNTCS** [comment]
> Please be careful about what you added in a commit.
> You must not add unrelated code change in a commit.
> 
> You can use `git add -p` command or CLion's tool to select only the diff you need to add in a commit.
> 
> These links will help you:
> https://gist.github.com/mattlewissf/9958704
> https://blog.jetbrains.com/idea/2018/02/intellij-idea-2018-1-eap-partial-git-commits-and-more/
> 
> Also, I wrote a help message in our wiki about this:
> https://github.com/kodebox-io/kodebox-internal/wiki/Tips-for-Developers#selectively-stage-your-work

File: test/package.json [hunk 7]
```diff
@@ -50,6 +50,7 @@
     "codechain-sdk": "^2.0.0-alpha.2",
     "codechain-stakeholder-sdk": "https://github.com/majecty/codechain-stakeholder-sdk-js#v2.0.0-alpha.1",
     "elliptic": "^6.4.1",
+    "foundry-rpc": "github:MSNTCS/foundry-rpc-js#alpha",
```
Comment (majecty) [inline test/package.json]: Let's use  `"foundry-rpc": "github:CodeChain-io/foundry-rpc-js#alpha` instead.

File: test/src/e2e/changeParams.test.ts [hunk 8]
```diff
@@ -220,13 +219,13 @@ describe("ChangeParams", function() {
                         fee: 10
                     })
             );
-            const pay = await node.sendPayTx({ seq: seq + 1, fee: 10 });
+            const pay = await node.sendPayTx({ seq: seq + 1, fee: 20 });
```
Comment (majecty) [inline test/src/e2e/changeParams.test.ts]: Why did you change the fee?

File: test/src/e2e/changeParams.test.ts [hunk 9]
```diff
@@ -429,7 +428,7 @@ describe("ChangeParams", function() {
         {
             await node.sdk.rpc.devel.stopSealing();
             const blockNumber = await node.sdk.rpc.chain.getBestBlockNumber();
-            const seq = await node.sdk.rpc.chain.getSeq(faucetAddress);
+            const seq = ( await node.rpc.chain.getSeq({address: faucetAddress.toString(), blockNumber:null}))!;
```
Comment (majecty) [inline test/src/e2e/changeParams.test.ts]: DId you run the formatter?
It seems that the formatter will remove the white space of `( await`.

File: test/src/e2e/customAction.test.ts [hunk 10]
```diff
@@ -42,27 +45,33 @@ describe("customAction", function() {
 
     describe("customAction", function() {
         it("should get initial state", async function() {
-            const actionData = await node.sdk.rpc.engine.getCustomActionData(
-                hitActionHandlerId,
-                ["hit count"]
+            const actionData = await node.rpc.engine.getCustomActionData({
+                    handlerId: hitActionHandlerId,
+                    bytes: `0xca${hitcount}`,
```
Comment (majecty) [inline test/src/e2e/customAction.test.ts]: Why the `ca` is needed? Do you know that?

Comment (majecty) [reply test/src/e2e/customAction.test.ts]: It seems that toHex(RLP.encode(["hit count"])) == `0xca${toHex(RLP.encode("hit count"))}`. Could you check it?

File: test/src/helper/spawn.ts [hunk 11]
```diff
@@ -45,6 +46,7 @@ const projectRoot = `${__dirname}/../../..`;
 export type SchemeFilepath = string;
 export type ChainType =
     | "solo"
+    | "simple_poa"
```
Comment (majecty) [inline test/src/helper/spawn.ts]: It should not included here.

File: test/src/helper/spawn.ts [hunk 12]
```diff
@@ -699,7 +717,7 @@ export default class CodeChain {
         tx: Transaction & AssetTransaction,
         options: { seq?: number } = {}
     ): Promise<H256> {
-        await this.sdk.rpc.devel.stopSealing();
+        await this.rpc.devel?.stopSealing();
```
Comment (majecty) [inline test/src/helper/spawn.ts]: Let's use `!` here. The devel should exist.

File: test/src/e2e/snapshot.test.ts [hunk 13]
```diff
@@ -41,9 +41,7 @@ describe("Snapshot", async function() {
 
         const blockHash = (await node.sdk.rpc.chain.getTransaction(pay.hash()))!
             .blockHash!;
-        await node.sdk.rpc.sendRpcRequest("devel_snapshot", [
-            blockHash.toJSON()
-        ]);
+        await node.rpc.devel?.developSnapshot({hash: blockHash.toJSON()})!;
```
Comment (majecty) [inline test/src/e2e/snapshot.test.ts]: Let's use `devel!`

File: test/src/e2e/account.test.ts [hunk 14]
```diff
@@ -19,6 +19,8 @@ import "mocha";
 import { invalidAddress, invalidSecret } from "../helper/constants";
 import { ERROR } from "../helper/error";
 import CodeChain from "../helper/spawn";
+import { H256, PlatformAddress } from "codechain-sdk/lib/core/classes";
+import { log } from "util";
```
Comment (MSNTCS) [inline test/src/e2e/account.test.ts]: They are mistakenly included. I will remove them.

File: test/src/helper/spawn.ts [hunk 15]
```diff
@@ -406,10 +418,21 @@ export default class CodeChain {
         if (!this.process) {
             return Promise.reject(Error("process isn't available"));
         }
-        await this.sdk.rpc.network.connect("127.0.0.1", peer.port);
+        await this.rpc.net.connect({ address: "127.0.0.1", port: peer.port });
+        await this.rpc.net.connect({ address: "127.0.0.1", port: peer.port });
```
Comment (majecty) [inline test/src/helper/spawn.ts]: @MSNTCS The line is the same as the above line. Please remove one.

File: test/src/e2e/customAction.test.ts [hunk 16]
```diff
@@ -78,59 +86,75 @@ describe("customAction", function() {
...
         it("should throw state not exist", async function() {
             try {
-                await node.sdk.rpc.engine.getCustomActionData(
-                    hitActionHandlerId,
-                    ["hit count"],
-                    99999
-                );
+                await node.rpc.engine.getCustomActionData({
+                    handlerId: hitActionHandlerId,
+                    bytes: `0xca${hitcount}`,
```
Comment (majecty) [inline test/src/e2e/customAction.test.ts]: Why do we need `0xca`?

File: test/src/e2e/customAction.test.ts [hunk 17]
```diff
@@ -78,59 +86,75 @@ describe("customAction", function() {
...
+                expect(e.toString()).include(ERROR.STATE_NOT_EXIST);
             }
         });
 
         it("should throw handler not found on getCustomActionData", async function() {
             try {
-                await node.sdk.rpc.engine.getCustomActionData(999999, []);
+                await node.rpc.engine.getCustomActionData({
+                    handlerId: 999999,
+                    bytes: "0xc0",
```
Comment (majecty) [inline test/src/e2e/customAction.test.ts]: What is "0xca"? Could you remove all the magic strings?

File: test/src/e2e/snapshot.test.ts [hunk 18]
```diff
@@ -41,9 +41,7 @@ describe("Snapshot", async function() {
 
         const blockHash = (await node.sdk.rpc.chain.getTransaction(pay.hash()))!
             .blockHash!;
-        await node.sdk.rpc.sendRpcRequest("devel_snapshot", [
-            blockHash.toJSON()
-        ]);
+        await node.rpc.devel?.developSnapshot({ hash: blockHash.toJSON() })!;
```
Comment (majecty) [inline test/src/e2e/snapshot.test.ts]: Please do not use `devel?` in all place.
